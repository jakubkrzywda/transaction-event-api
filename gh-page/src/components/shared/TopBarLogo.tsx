import React from "react";
import logo from "../../img/logo.svg";

const TopBarLogo = () => {
    return <div className="logo"><img src={logo} alt="Logo" height="90" /></div>;
};

export default TopBarLogo;