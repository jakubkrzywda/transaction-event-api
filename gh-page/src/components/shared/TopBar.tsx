import React from "react";
import TopBarLogo from "./TopBarLogo";

const TopBar = () => {
    return (
        <div className="app-header">
            <TopBarLogo />
        </div>
    );
};

export default TopBar;