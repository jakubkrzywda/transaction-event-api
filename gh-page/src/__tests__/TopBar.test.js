import React from "react";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import TopBar from "../components/shared/TopBar";
import TopBarLogo from "../components/shared/TopBarLogo";

Enzyme.configure({adapter: new Adapter()});

describe("TopBar", () => {
    it("renders without crashing", () => {
        const wrapper = shallow(<TopBar />);
        expect(wrapper.exists()).toBe(true);
    });
  
    it("matches snapshot", () => {
        const wrapper = shallow(<TopBar />);
        expect(wrapper).toMatchSnapshot();
    });
    
    it("has correct class name", () => {
        const wrapper = shallow(<TopBar />);
        expect(wrapper.hasClass("app-header")).toBe(true);
    });
    
    it("renders TopBarLogo", () => {
        const wrapper = shallow(<TopBar />);
        expect(wrapper.find(<TopBarLogo/>)).toBeDefined();
    });
});