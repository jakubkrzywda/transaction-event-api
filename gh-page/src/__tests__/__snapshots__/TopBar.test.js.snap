// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar matches snapshot 1`] = `ShallowWrapper {}`;
